woocommerce/woocommerce-gutenberg-products-block
# Related Products: hide the block outside of Single Product Template and Single Product block

## What

There was a bug that Related Products block was available in the inserter outside of Single Product template which caused a fatal error when trying to save the template. Related Product is supposed to be used only within Single Product template and Single Product block.

This PR improves the logic of [`registerBlockSingleProductTemplate`](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/assets/js/atomic/utils/register-block-single-product-template.ts#L26C14-L26C48) function.

Fixes https://github.com/woocommerce/woocommerce-blocks/issues/10959

## Why

Changes:
- Removing the `ancestor` parameter from `registerBlockVariation` function since variation doesn't support `ancestor` property. This change doesn't influence the logic, but is added to decrease complexity of the function.

- Provide the default value of `ancestor` field when registering the blocks via `registerBlockSingleProductTemplate`. As the function name suggests the blocks should only be available within the `single-product` template or block. While block didn't provide this property it was actually registered globally as it happened in case of Related Products block. So now if the block doesn't provide `ancestor` field we fallback to `ancestor: [ 'woocommerce/single-product' ]`.



## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Go to Single Product template
2. Try to add "Related Products" and "Product Reviews" blocks
3. **Expected:** they should be available in the inserter and should be added normally
4. Save and go to frontend
5. Verify there's no fatal error

---

1. Go to Product Catalog template
2. Try to add "Related Products" and "Product Reviews" blocks
3. **Expected:** they **should NOT** be available in inserter
4. Add Single Product block
5. Try to add "Related Products" and "Product Reviews" in scope of Single Product block
6. **Expected:** they should be available in the inserter and should be added normally
7. Save and go to frontend
8. Verify there's no fatal error

---

1. Create new post
2. Try to add "Related Products" and "Product Reviews" blocks
3. **Expected:** they **should NOT** be available in inserter
4. Add Single Product block
5. Try to add "Related Products" and "Product Reviews" in scope of Single Product block
6. **Expected:** they **should NOT** be available in inserter



## Screenshots or screencast



| Before | After |
| ------ | ----- |
|        |       |

## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog


> Related Products: hide the block outside of Single Product Template and Single Product block
